### ISSUE TITLE
How can I get it to maximum speed and quality for Z-Image Lora training with Musubi Tuner?

### ISSUE THREAD

**Issue by peepeepeepoopoopoo** [kohya-ss/musubi-tuner#788]
**I'm using a B200 (CUDA 12.9, PyTorch 2.9.1, Python 3.12, Linux).**

Can make 200/200 [00:47<00:00,  4.23it/s, avr_loss=0.43]

I've tried many methods, but the VRAM usage won't go above 70GB. I want maximum speed and quality, I would greatly appreciate your advice.. Forgive my ignorance, but I know more about practical than theoretical aspects of AI.

-

**I work this:**

#2.2 (CREATE THE DATASET.TOML FILE)

cat > /workspace/musubi-tuner/dataset/dataset.toml << 'EOF'
# resolution, caption_extension, batch_size, num_repeats, enable_bucket, bucket_no_upscale should be set in either general or datasets
# otherwise, the default values will be used for each item
# general configurations
[general]
resolution = [960 , 960]
caption_extension = ".txt"
batch_size = 32
enable_bucket = true
bucket_no_upscale = false

[[datasets]]
image_directory = "/workspace/musubi-tuner/dataset"
cache_directory = "/workspace/musubi-tuner/dataset/cache"
num_repeats = 1 # optional, default is 1. Number of times to repeat the dataset. Useful to balance the multiple datasets with different sizes.
# other datasets can be added here. each dataset can have different configurations
EOF

#Verify that it was created correctly
echo "Archivo creado en:"
ls -la /workspace/musubi-tuner/dataset/dataset.toml

-

python src/musubi_tuner/zimage_cache_latents.py --dataset_config /workspace/musubi-tuner/dataset/dataset.toml --vae /workspace/musubi-tuner/models/vae/chroma_flux_z_image/ae.safetensors
python src/musubi_tuner/zimage_cache_text_encoder_outputs.py --dataset_config /workspace/musubi-tuner/dataset/dataset.toml --text_encoder /workspace/musubi-tuner/models/clip/z_image/qwen_3_4b.safetensors

accelerate launch --mixed_precision bf16 --dynamo_backend inductor --num_processes 1 --num_machines 1 --num_cpu_threads_per_process 8 src/musubi_tuner/zimage_train_network.py --dit /workspace/ComfyUI/models/diffusion_models/z_image/z_image_de_turbo_v1_bf16.safetensors --vae /workspace/musubi-tuner/models/vae/chroma_flux_z_image/ae.safetensors --text_encoder /workspace/musubi-tuner/models/text_encoders/z_image/qwen_3_4b.safetensors --dataset_config /workspace/musubi-tuner/dataset/dataset.toml --flash-attn --timestep_sampling shift --weighting_scheme none --discrete_flow_shift 2.0 --optimizer_type adamw --learning_rate 3e-4 --max_data_loader_n_workers 48 --persistent_data_loader_workers --network_module networks.lora_zimage --network_dim 128 --max_train_epochs 200 --save_every_n_epochs 1 --seed 42 --output_dir /workspace/musubi-tuner/output --output_name z_image_lora_b200_extreme

for f in /workspace/musubi-tuner/output/*; do b=$(basename "$f"); d=/workspace/ComfyUI/models/loras/$b; [ ! -e "$d" ] && mv "$f" "$d" || i=1; while [ -e "/workspace/ComfyUI/models/loras/${b%.*}_$i.${b##*.}" ]; do ((i++)); done; mv "$f" "/workspace/ComfyUI/models/loras/${b%.*}_$i.${b##*.}"; done

-

**log: https://pastebin.com/SKkJ0Sb8**
